frontend/public/co-fetch.js

```diff
@@ -109,12 +109,19 @@ export const coFetch = (url, options = {}, timeout = 20000) => {
     delete allOptions.headers['X-CSRFToken'];
   }
 
+  // return fetch promise directly if timeout <= 0
+  if (timeout < 1) {
+    return fetch(url, allOptions).then((response) => validateStatus(response, url));
+  }
```

Comment:
You can remove some duplicate code by storing the result of `fetch` in a var. Then use the same var in `Promise.race`.
```suggestion
  const fetchPromise = fetch(url, allOptions).then((response) => validateStatus(response, url));
  if (timeout < 1) {
    return fetchPromise;
  }
```

Reply:
Yeah, that was exactly what I did at first. But then realized that promises are not lazy and I would be executing the `fetchPromise` as soon as I do a `.then` on it. Not that it would affect anything seriously but wasn't sure if that was the correct way.

Reply:
Updated.